students/YingGuo/session05/mailroom3.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+"""session05, assignment mailroom3, Ying Guo"""
+
+from collections import defaultdict
+
+data =defaultdict(list, {"Tom":[100,200], "Tim":[3000], "Ken": [888,888,888], "Ted":[8000,5000,500], "Jon":[1000,6000]})
+
+#Send a thank you function 
+def thank_you(data):
+    """record a new donation and send thank you letter to the donator"""
+    while True:
+        try:
+            user_input = input("please enter the name of donor. You can enter list to see the exisitng list of donors. Or enter quit to exist current function")
+            if user_input.lower() == "list":
+                print(data.keys())
+
+            elif user_input.lower() == "quit":
+                break
+            
+            else:
+                amount1 = int(input("how much would you like to donat?"))
+                data[user_input.capitalize()] += [amount1]
+                print("Thank you {} for your donation {}!".format(user_input.capitalize(), amount1))
+                return data
+        except ValueError:
```

Comment:
I think this will work, but it looks to me the scope of the try-except block is unnecessarily wider than necessary. I believe the `ValueError` is for line 22 (`int(input(...))`) only, and it's best to narrow down the try-except scope to the needed code only.